packages/init/src/SliceMachineInitProcess.ts

```diff
@@ -0,0 +1,1114 @@
+
+${chalk.cyan("?")} Your Prismic repository name`.replace("\n", ""),
+				);
+			},
+			validate: async (rawDomain: string) => {
+				const domain = formatRepositoryDomain(rawDomain);
+				const validation = await validateRepositoryDomainAndAvailability({
+					domain,
+					existsFn: (domain) =>
+						this.manager.prismicRepository.checkExists({ domain }),
+				});
+				const errorMessage = getErrorMessageForRepositoryDomainValidation({
+					validation,
+					displayDomain: chalk.cyan(domain),
+				});
+
+				return errorMessage || true;
+			},
+			format: (value) => {
+				return formatRepositoryDomain(value);
+			},
+		});
+
+		// Clear extra lines
+		process.stdout.moveCursor?.(0, -16);
+		process.stdout.clearScreenDown?.();
+
+		this.context.repository = {
+			domain,
+			exists: false,
+		};
+	}
+
+	protected createNewRepository(): Promise<void> {
+		assertExists(
+			this.context.repository,
+			"Repository selection must be available through context to run `createNewRepository`",
+		);
+
+		return listrRun([
+			{
+				title: `Creating new repository ${chalk.cyan(
+					this.context.repository.domain,
+				)} ...`,
+				task: async (_, task) => {
+					assertExists(
+						this.context.repository,
+						"Repository selection must be available through context to run `createNewRepository`",
+					);
+					assertExists(
+						this.context.framework,
+						"Project framework must be available through context to run `createNewRepository`",
+					);
+
+					await this.manager.prismicRepository.create({
+						domain: this.context.repository.domain,
+						framework: this.context.framework.prismicName,
+					});
+
+					this.context.repository.exists = true;
+					task.title = `Created new repository ${chalk.cyan(
+						this.context.repository.domain,
+					)}`;
+				},
+			},
+		]);
+	}
+
+	protected finishCoreDependenciesInstallation(): Promise<void> {
+		return listrRun([
+			{
+				title: `Finishing core dependencies installation with ${chalk.cyan(
+					this.context.packageManager,
+				)} ...`,
+				task: async (_, task) => {
+					assertExists(
+						this.context.installProcess,
+						"Initial dependencies installation process must be available through context to run `finishCoreDependenciesInstallation`",
+					);
+
+					const updateOutput = (data: Buffer | null) => {
+						if (data instanceof Buffer) {
+							task.output = data.toString();
+						}
+					};
+
+					// Don't clutter console with logs when process is non TTY (CI, etc.)
+					if (process.stdout.isTTY || process.env.NODE_ENV === "test") {
+						this.context.installProcess.stdout?.on("data", updateOutput);
+					}
+					this.context.installProcess.stderr?.on("data", updateOutput);
+
+					try {
+						await this.context.installProcess;
+					} catch (error) {
+						/**
+						 * Error catching happens when then process is started ealier so
+						 * that all install errors, earlier and presents, can be catched.
```

Comment:
Can you rephrase this?

Reply:
I changed the comment to this:

```ts
/**
 * Be aware that `this.context.installProcess` is an Execa Promise
 * that has a `.catch()` Promise handler defined when it was
 * created. This additional `catch` block handles a case where an
 * error is not caught in time or is thrown outside the Execa
 * process.
 *
 * We force the current task to wait a few extra seconds to allow
 * the previoulsy defined `.catch()` handler to run.
 *
 * If the `.catch()` handler wasn't able to run in time, we re-throw
 * the error and let a parent context handle it.
 */
```

@lihbr When you're back at work, could you verify if this updated description is accurate? I don't know the exact purpose of the extra `try...catch` considering we already have a `.catch()`, so I wrote my best guess.